CIP-transaction-inputs-as-unordered-set/README.md

```diff
@@ -0,0 +1,140 @@
+---
+CIP: xxx
+Title: Transaction Inputs as Unordered Set
+Category: Plutus
+Status: Proposed
+Authors:
+  - Jonathan Rodriguez <info@anastasialabs.com>
+Implementors: []
+Discussions:
+  - https://github.com/cardano-foundation/CIPs/pull/?
+Created: 2024-02-01
+License: CC-BY-4.0
+---
+
+
+## Abstract
+
+We propose the introduction of a new structure for transaction inputs aimed at enhancing the execution efficiency of Plutus contracts.
+
+This CIP facilitates the preservation of input ordering from a submitted transaction, diverging from the current state where the ledger reorders transaction inputs lexicographically. Maintaining the input ordering from a submitted transaction enables a validator to efficiently identify the required inputs in accordance with its validation logic, thus eliminating the need to use unnecessary computation by traversing all transaction inputs or performing sorting within the validator itself.
+
+Furthermore, this implementation offers the potential to enhance existing design patterns already used in production, specifically by improving the ability to group and match inputs and outputs located at the specific index positions.
+
+## Motivation: why is this CIP necessary?
+
+According to the Babbage CDDL [transaction_body](https://github.com/IntersectMBO/cardano-ledger/blob/master/eras/babbage/impl/cddl-files/babbage.cddl) , the inputs and reference inputs of a transaction body are represented as a set, which indicates the non-duplicate inputs. However, the ledger not only process the [transaction_inputs](https://github.com/IntersectMBO/cardano-ledger/blob/0274cf65dbb79773122b69dfd36a8299eec2783f/eras/babbage/impl/cddl-files/babbage.cddl#L75-L77) as a set , but also orders them lexicographically, first by `transaction_id` and then by `index`.
```

Comment:
```suggestion
According to the Babbage CDDL [transaction_body](https://github.com/IntersectMBO/cardano-ledger/blob/master/eras/babbage/impl/cddl-files/babbage.cddl) , the inputs and reference inputs of a transaction body are represented as a set, which indicates the non-duplicate inputs. However, the ledger not only process the [transaction_inputs](https://github.com/IntersectMBO/cardano-ledger/blob/0274cf65dbb79773122b69dfd36a8299eec2783f/eras/babbage/impl/cddl-files/babbage.cddl#L75-L77) as a set , but also orders them lexicographically, first by `transaction_id` and then by `index`.
```
maybe the CDDL should also reflect that is is a non_empty_set 🤔

Reply:
CDDL does not prevent for inputs from being empty. It is the ledger rules that enforce this